Transfer/Sources/TransferAmountCell.swift

```diff
@@ -109,41 +109,71 @@ final class TransferAmountCell: UITableViewCell {
 
 extension TransferAmountCell: UITextFieldDelegate {
     func textFieldDidEndEditing(_ textField: UITextField) {
-        enteredAmountHandler?(amountTextField.text ?? "")
+        guard var currentText = textField.text else {
+            enteredAmountHandler?("")
+            return
+        }
+        let decimalPointSymbol = "."
+        if currentText.last == Character(decimalPointSymbol) {
+            currentText.removeLast()
+            textField.text = currentText
+        }
+        enteredAmountHandler?(currentText)
     }
 
     func textField(_ textField: UITextField,
                    shouldChangeCharactersIn range: NSRange,
                    replacementString string: String) -> Bool {
-        var currentText = textField.text ?? ""
+        let maximumIntegerDigits = 12
+        let maximumFractionDigits = 2
+        let decimalPointSymbol = "."
+        let groupSymbol = ","
+        let currentText = textField.text ?? ""
+
+        // BackSpace
+        if string.isEmpty {
+            return true
+        }
+        // Paste
         if string.count > 1 {
-            // Paste
             let stringPastedAmount = string.format(with: currencyLabel.text)
+            let pastedDigitsOnly = stringPastedAmount.replacingOccurrences(of: "[^0-9]",
+                                                                           with: "",
+                                                                           options: .regularExpression)
+            let pointEntered = stringPastedAmount.contains(decimalPointSymbol)
+            let maximumAllowedDigits = pointEntered
+                ? maximumIntegerDigits + maximumFractionDigits
+                : maximumIntegerDigits
             guard !stringPastedAmount.isEmpty,
-                currentText.isEmpty else {
+                currentText.isEmpty,
+                pastedDigitsOnly.count <= maximumAllowedDigits else {
                     return false
             }
-            textField.text = stringPastedAmount
+            textField.text = stringPastedAmount.replacingOccurrences(of: groupSymbol, with: "")
```

Comment:
No need to remove group symbol. Our API accepts amount like 1,111 so it's ok to have , in the amount field if user enters it.

Reply:
user can delete the last "1" and send "1,11" - it is wrong. That's why the user should do an additional action to remove ","